viewing_party/party.py

```diff
@@ -1,23 +1,206 @@
+    rec_movies = []
+    for movies in friends_unique_movies:
+        if movies['host'] in user_data['subscriptions'] and movies not in user_data["watched"]:
+            rec_movies.append(movies)
+            
+    return rec_movies
+
+
+
+        
+        #friends movies - user movie and host in user sub= rec movies
+        
+    
+
+# def get_friend_movie_list(user_data):
+#     friend_movie_list = []
+#     for i in range(len(user_data["friends"])):
+#         for j in range (len(user_data["friends"][i]["watched"])):
+#             friend_movie = user_data["friends"][i]["watched"][j]
+#             friend_movie_list.append(friend_movie) 
+            
+            
+# def get_user_movie_list(user_data):
+#     user_movie_list = []
+#     for i in range (len(user_data["watched"])):
+#         user_movie = user_data["watched"][i]
+#         user_movie_list += user_movie_list.append(user_movie) 
+
+# def get_available_recs(user_data):
+#     recommended_movies = []
+#     for movie in get_friend_movie_list(user_data):
+#         if movie not in get_user_movie_list(user_data) and get_friend_movie_list["host"] == user_data["subscriptions"]:
+#             recommended_movies.append(movie)
+
+            #if movie not in user_movie_list and in friend_movie_list:
+            #if user_data["subscriptions"] == friend_movie_list["host"]:
+#user_data ---- subscriptions (list of strings (streaming service))
+#friend---- friends ----watched list - movie - host
+#   - `user_data` will have a field `"subscriptions"`. The value of `"subscriptions"` is a list of strings
+#     - This represents the names of streaming services that the user has access to
+#     - Each friend in `"friends"` has a watched list. Each movie in the watched list has a `"host"`,
+#  which is a string that says what streaming service it's hosted on
+# - Determine a list of recommended movies. A movie should be added to this list if and only if:
+#   - The user has not watched it
+#   - At least one of the user's friends has watched
+#   - The `"host"` of the movie is a service that is in the user's `"subscriptions"`
+# - Return the list of recommended movies
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+
+    rec_list = []
+    user_genre = get_most_watched_genre(user_data)
+    friends_unique_movies = get_friends_unique_watched(user_data)
+
+    for movies in friends_unique_movies:
+        if movies['genre'] == user_genre:
+            rec_list.append(movies)
+    
+    return rec_list
+
+def get_rec_from_favorites(user_data):
+    fav_rec_list = []
+    friends_unique_movies = get_friends_unique_watched(user_data)
+
+    for movie in get_unique_watched(user_data):
+        if movie not in friends_unique_movies:
+            fav_rec_list.append(movie)
+    print(fav_rec_list)
+    return fav_rec_list

```

Comment:
This implementation is passing the test `test_unique_rec_from_favorites`, but there's an issue. This approach will return every item in `get_unique_watched(user_data)` whether or not they are in the list `user_data['favorites']`. How could you update this function to consider the user's favorite list?